pkg/gofr/gofr.go

```diff
@@ -374,11 +351,50 @@ func (a *App) getExporter(name, host, port, url string) (sdktrace.SpanExporter,
 
 		exporter = NewExporter(url, logging.NewLogger(logging.INFO))
 	default:
-		a.container.Errorf("unsupported trace exporter: %s", name)
+		if name == "" {
+			a.container.Errorf("TRACE_EXPORTER env is missing")
+		} else {
+			a.container.Errorf("unsupported trace exporter: %s", name)
+		}
 	}
 
 	return exporter, err
 }
+func (a *App) buildOtlp(url, host, port, exporter string) (sdktrace.SpanExporter, error) {
+	if url == "" {
+		url = fmt.Sprintf("%s:%s", host, port)
+	}
+
+	a.container.Logf("Exporting traces to %s at %s", exporter, url)
+
+	opts := []otlptracegrpc.Option{otlptracegrpc.WithInsecure(), otlptracegrpc.WithEndpoint(url)}
+
+	authHeader := a.Config.Get("TRACER_AUTH_KEY")
+
+	if authHeader != "" {
+		opts = append(opts, otlptracegrpc.WithHeaders(map[string]string{"Authorization": authHeader}))
+	}
+
+	return otlptracegrpc.New(context.Background(), opts...)
```

Comment:
I would expect the current `buildOtlp` (whatever its future would be)  to receive the context as argument.

It's always a bad thing when a call to `context.Background()` is made in the depth of the stack.

I know, you are refactoring by moving code, but here you moved the context creation to a lower level, it should be avoided